src/erasure-code/clay/ErasureCodePluginClay.cc

```diff
@@ -40,5 +40,10 @@ const char *__erasure_code_version() { return CEPH_GIT_NICE_VER; }
 int __erasure_code_init(char *plugin_name, char *directory)
 {
   auto& instance = ceph::ErasureCodePluginRegistry::instance();
-  return instance.add(plugin_name, new ErasureCodePluginClay());
+  auto plugin = std::make_unique<ErasureCodePluginClay>();
+  int r = instance.add(plugin_name, plugin);
```

Comment:
```suggestion
  int r = instance.add(plugin_name, plugin.get());
```

sorry, i missed this in the previous comment .

Reply:
Will have a try, thanks~